### PR TITLE
mbp2018-bridge-drv: init at 0.01

### PR DESCRIPTION
**Pull request by hlolli**
###### Motivation for this change

This kernel-module is part of macbook pro 15+ hardware support https://github.com/NixOS/nixos-hardware/pull/231

###### Things done

Adding this kernel-module under mbp-modules, as I intend to add at least 1 more macbook pro kernel module. But this nesting is perhaps debateable. Also the author hasn't made any releases but I did find some metadata in the c file https://github.com/MCMrARM/mbp2018-bridge-drv/blob/master/pci.c#L431-L434 so I'm assuming gpl2 license.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  src = fetchFromGitHub {
+    owner = "MCMrARM";
+    repo = "mbp2018-bridge-drv";
+    rev = "0.01";
+    sha256 = "0ac2l51ybfrvg8m36x67rsvgjqs1vwp7c89ssvbjkrcq3y4qdb53";
+  };
+
+  buildPhase = ''
+    make -C ${kernel.dev}/lib/modules/${kernel.modDirVersion}/build \
+      M=$(pwd) modules
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix:16]: I think ``-j$NIX_BUILD_CORES`` is missing.

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      INSTALL_MOD_PATH=$out M=$(pwd) modules_install
+  '';
+
+  meta = with lib; {
+    description = "A driver for MacBook models 2018 and newer, which makes the keyboard, mouse and audio output work.";
+    longDescription = ''
+      A driver for MacBook models 2018 and newer, implementing the VHCI (required for mouse/keyboard/etc.) and audio functionality.
+    '';
+    homepage = "https://github.com/MCMrARM/mbp2018-bridge-drv";
+    license = lib.licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix:30]: gpl2Only or gpl2Plus?

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    platforms = platforms.linux;
+    maintainers = [ ];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix:32]: Would you mind adding yourself as a maintainer?

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -19245,6 +19245,8 @@ in
 
     tbs = callPackage ../os-specific/linux/tbs { };
 
+    mbp2018-bridge-drv = callPackage ../os-specific/linux/mbp-modules/mbp2018-bridge-drv { inherit kernel; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:19248]:
```suggestion
    mbp2018-bridge-drv = callPackage ../os-specific/linux/mbp-modules/mbp2018-bridge-drv { };
```

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, kernel, fetchFromGitHub, }:
+
+stdenv.mkDerivation rec {
+  name = "mbp2018-bridge-drv";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix:4]:
```suggestion
  pname = "mbp2018-bridge-drv";
```

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix (hunk 2 above)
Comment (hlolli) [reply pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix:30]: Not sure about that, the author includes `MODULE_LICENSE("GPL");` And I think there's specific criteria under which the linux kernel is link-able?

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix (hunk 1 above)
Comment (hlolli) [reply pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix:16]: wont hurt to add it, but maybe it's already declared with the kernel module wrapper, not sure. Either way, adding it...

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix (hunk 3 above)
Comment (hlolli) [reply pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix]: done

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix (hunk 1 above)
Comment (hlolli) [reply pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix]: done

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix (hunk 2 above)
Comment (Mic92) [reply pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix:30]: The linux kernel is gpl2Only so its probably save to assume this module is too.

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, kernel, fetchFromGitHub, }:
+
+stdenv.mkDerivation rec {
+  pname = "mbp2018-bridge-drv";
+  version = "b43fcc069da73e051072fde24af4014c9c487286";
```
Comment (Mic92) [inline pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix:5]:
```suggestion
  version = "0.01";
```

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchFromGitHub {
+    owner = "MCMrARM";
+    repo = "mbp2018-bridge-drv";
+    rev = "0.01";
```
Comment (Mic92) [inline pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix:10]:
```suggestion
    rev = "${version}";
```

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix (hunk 2 above)
Comment (hlolli) [reply pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix]: ok version fixed, for the other rev/version mixup, unforgivable :facepalm:

**Comment by hlolli** [comment]
One thing I think is missing @Mic92 is `hardeningDisable = [ "pic" "format" ];` going to test here locally if that was a factor. Certainly doesn't prevent it from being installed. But it was listed as needed in the kernel section.

**Comment by hlolli** [comment]
hardeningDisable doesn't matter at all. Tested it and even restarted, works all the same :+1: lgtm @Mic92

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 113093` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.mbp2018-bridge-drv</li>
    <li>linuxPackages_hardkernel_4_14.mbp2018-bridge-drv</li>
    <li>linuxPackages_hardkernel_latest.mbp2018-bridge-drv</li>
    <li>linuxPackages_latest-libre.mbp2018-bridge-drv</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>linuxPackages_4_14.mbp2018-bridge-drv</li>
    <li>linuxPackages_4_19.mbp2018-bridge-drv</li>
    <li>linuxPackages_4_4.mbp2018-bridge-drv</li>
    <li>linuxPackages_4_9.mbp2018-bridge-drv</li>
  </ul>
</details>
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>linuxPackages.mbp2018-bridge-drv (linuxPackages_5_4.mbp2018-bridge-drv)</li>
    <li>linuxPackages_5_10.mbp2018-bridge-drv (linuxPackages_latest.mbp2018-bridge-drv)</li>
    <li>linuxPackages_hardened.mbp2018-bridge-drv</li>
    <li>linuxPackages_latest_hardened.mbp2018-bridge-drv</li>
    <li>linuxPackages_latest_xen_dom0.mbp2018-bridge-drv</li>
    <li>linuxPackages_latest_xen_dom0_hardened.mbp2018-bridge-drv</li>
    <li>linuxPackages_lqx.mbp2018-bridge-drv</li>
    <li>linuxPackages_testing_bcachefs.mbp2018-bridge-drv</li>
    <li>linuxPackages_xen_dom0.mbp2018-bridge-drv</li>
    <li>linuxPackages_xen_dom0_hardened.mbp2018-bridge-drv</li>
    <li>linuxPackages_zen.mbp2018-bridge-drv</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
> 4 packages failed to build:
> * linuxPackages_4_14.mbp2018-bridge-drv
> * linuxPackages_4_19.mbp2018-bridge-drv
> * linuxPackages_4_4.mbp2018-bridge-drv
> * linuxPackages_4_9.mbp2018-bridge-drv

The module most likely needs to be disabled/marked broken for those kernels.

File: pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchFromGitHub {
+    owner = "MCMrARM";
+    repo = "mbp2018-bridge-drv";
+    rev = "${version}";
+    sha256 = "0ac2l51ybfrvg8m36x67rsvgjqs1vwp7c89ssvbjkrcq3y4qdb53";
+  };
+
+  buildPhase = ''
+    make -C ${kernel.dev}/lib/modules/${kernel.modDirVersion}/build \
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix:15]: Could makeFlags be used here?

Comment (Mic92) [reply pkgs/os-specific/linux/mbp-modules/mbp2018-bridge-drv/default.nix:15]: No because installPhase has different flags.

**Pull request merged**